specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,1026 @@
+
+    // Assign initial values for first toast progress UI
+    ToastProgressData data;
+    data.Title(L"Weekly playlist"); // Binds to {progressTitle} in xml payload
+    data.Value(0.6); // Binds to {progressValue} in xml payload
+    data.ValueStringOverride(L"15/26 songs"); // Binds to {progressValueString} in xml payload
+    data.Status(L"Downloading..."); // Binds to {progressStatus} in xml payload
+    data.SequenceNumber(1);
+
+    toast.ProgressData(data);
+    ToastNotificationManager::Default().ShowToast(toast);
+}
+
+// Send subsequent progress updates
+winrt::Windows::Foundation::IAsyncAction UpdateProgressAsync()
+{
+    winrt::hstring tag = L"weekly-playlist";
+    winrt::hstring group = L"downloads";
+
+    // Assign new values
+    // Note that you only need to assign values that changed. In this example we don't assign progressStatus since we don't need to change it
+    ToastProgressData data;
+    data.Value(0.7); // Binds to {progressValue} in xml payload
+    data.ValueStringOverride(L"18/26 songs"); // Binds to {progressValueString} in xml payload
+    data.SequenceNumber(2);
+
+    auto result = co_await ToastNotificationManager::Default().UpdateToastProgressDataAsync(data, tag, group);
+    if (result != ToastProgressResult::Succeeded)
+    {
+        LOG_HR_MSG(E_UNEXPECTED, "Toast Progress Update Failed!");
+    }
+}
+```
+
+# Remarks
+
+## Registration
+
+The developer should always call the Toast Registration API first to register the current process as
+the Activator target.
+
+## Foreground API calls
+
+The developer should always subscribe to toast activation foreground events to intercept toasts if
+the app happens to be running in the foreground.
+
+## Manifest Registration
+
+For MSIX, the COM activator GUID and the exe need to be registered in the manifest. The launch args
+would need to be pre-set to a well-known string that defines Toast Activation Triggers.
+
+```xml
+<Extensions>
+  <com:Extension Category="windows.comServer">
+    <com:ComServer>
+      <com:ExeServer Executable="SampleBackgroundApp\SampleBackgroundApp.exe" DisplayName="SampleBackgroundApp" Arguments="----ToastActivationServer">
+        <com:Class Id="{GUIDEntryPoint}" />
+      </com:ExeServer>
+    </com:ComServer>
+  </com:Extension>
+</Extensions>
+```
+
+# API Details
+
+```c#
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.Windows.ToastNotifications
+{
+    [experimental]
+    // The Shell asset details for Unpackaged App Registrations
+    runtimeclass ToastAssets
+    {
+        // Initialize using Shell assets like DisplayName and iconPath
+        ToastAssets(String displayName, Windows.Foundation.Uri iconPath);
+
+        // The App friendly DisplayName for the toast in Action Centre
+        String DisplayName { get; };
+
+        // The full file path for the icon image
```

Comment:
can this be an http uri, or just a file / ms-appx kind of path?

Reply:
It needs to be a file path for unpackaged applications. So something like C:\test.png

Reply:
ms-appx is not applicable to unpackaged apps unfortunately.

Reply:
this API supports both unpackaged and packaged apps, though, right? In which case, ms-appx would/should still work?
Using Uri as the type, sort of implies that this is supported, as well as being able to use http/https links

Reply:
For packaged Apps, we expect the developer to call into: 
static ToastActivationInfo CreateFromActivationGuid(Guid taskClsid);
to retrieve the ToastInfo construct.

For unpackaged, we want the developer to call into:
static ToastActivationInfo CreateFromToastAssets(ToastAssets assets);
to retrieve the ToastInfo construct

The reason for this is that packaged apps already have their assets registered in their app manifest. An unpackaged app like notepad.exe doesn't necessarily have this and has to provide the assets programatically.

Reply:
That call is followed by RegisterActivator(ToastActivationInfo details); which is true for both Packaged and Unpackaged applications.